### ISSUE TITLE
Calculate each row's precision of a matrix

### ISSUE THREAD

**Issue by jasonshere** [tensorflow/addons#2257]
**tf.keras.metrics.Precision() will calculate the precision at top k, which returns a scalar value even if it's passed matrices as inputs. Is it better if it returns a 1-D tensor when using a 2-D matrix as inputs?**

E.g. tf.keras.metrics.Precision(top_k=3)([[1, 0, 1, 0], [0, 0, 1, 1]], [[0, 0, 1, 0], [0, 0, 1, 1]]) = 0.6666667
I think it's better to return a tensor [0.3333333, 1].

**Relevant information**
- Are you willing to contribute it (yes):
- Are you willing to maintain it going forward? (yes):
- Is there a relevant academic paper? (no):
- Is there already an implementation in another framework? (not sure):
- Was it part of tf.contrib? (no):

**Which API type would this fall under (layer, metric, optimizer, etc.)**
metric

**Who will benefit with this feature?**
Anyone

**Comment by seanpmorgan** [comment]
TensorFlow Addons is transitioning to a minimal maintenance and release mode. New features will not be added to this repository. For more information, please see our public messaging on this decision:
[TensorFlow Addons Wind Down](https://github.com/tensorflow/addons/issues/2807)

Please consider sending feature requests / contributions to other repositories in the TF community with a similar charters to TFA:
[Keras](https://github.com/keras-team/keras)
[Keras-CV](https://github.com/keras-team/keras-cv)
[Keras-NLP](https://github.com/keras-team/keras-nlp)
